src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(last, value, null);
+        if (isEmpty()) {
+            first = node;
+        } else {
+            last.next = node;
+        }
+        last = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size && checkIndex(index)) {
+            add(value);
+            return;
+        }
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
```

давай вернем как было, моя ошибка